core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/preferences/TitleBlockPreferencePage.java

```diff
@@ -0,0 +1,496 @@
+          if (content == "content") {
+            dialog.setCurrentName("");
+            dialog.setCurrentContent("");
+          } else {
+            dialog.setCurrentName(name);
+            dialog.setCurrentContent(content);
+          }
+          dialog.create();
+          if (dialog.open() == Window.OK) {
+            list.get(index).name = dialog.getName();
+            list.get(index).content = dialog.getContent();
+            cell.setText(dialog.getName());
+
+          }
+
+        }
+        }
+
+      }
+    };
+
+    v.getTable().addListener(SWT.MouseDoubleClick, treeListener);
+  }
+
+  private void disposeColumns() {
+    TableColumn[] columns = v.getTable().getColumns();
+    for (TableColumn tc : columns) {
+      tc.dispose();
+    }
+  }
+
+  private void refreshTableColumns() {
+    disposeColumns();
+    createColumns(v, columnsNumber);
+    v.setInput(updateModel(columnsNumber, rowsNumber));
+
+  }
+
+  private void refreshTableRows() {
+    rowsNumber += 1;
+    v.setInput(updateModelRows(columnsNumber, rowsNumber));
+  }
+
+  @Override
+  public boolean performOk() {
+
+    doGetPreferenceStore().setValue("defaultTitleBlock", defaultTitleBlockFieldEditor.getBooleanValue());
+    doGetPreferenceStore().setValue("columnFieldTitleBlock", columnsNumber);
+    doGetPreferenceStore().setValue("rowFieldTitleBlock", rowsNumber);
+    String s = "";
+    for (int i = 0; i < rowsNumber; i++) {
+      for (int j = 0; j < columnsNumber; j++) {
+        if (i == rowsNumber - 1 && j == columnsNumber - 1) {
+          s = s + tccMatrix.get(i).get(j).name + "+" + tccMatrix.get(i).get(j).content;
+        } else {
+          s = s + tccMatrix.get(i).get(j).name + "+" + tccMatrix.get(i).get(j).content + "+";
+
+        }
+
+      }
+    }
+    doGetPreferenceStore().setValue("tableTitleBlock", s);
+    return super.performOk();
+  }
+
+  private List<List<TitleBlockCell>> updateModel(int nrCol, int nrRows) {
+    for (int i = 0; i < nrRows; i++) {
+      List<TitleBlockCell> tbcCell = tccMatrix.get(i);
+      tbcCell.add(new TitleBlockCell("Edit", "aql:edit"));
+    }
+    return tccMatrix;
+  }
+
+  private List<List<TitleBlockCell>> updateModelRows(int nrCol, int nrRows) {
+    List<TitleBlockCell> tbcCell = new ArrayList<>();
+    for (int i = 0; i < nrCol; i++) {
+      tbcCell.add(new TitleBlockCell("Edit", "aql:edit"));
+    }
+    tccMatrix.add(tbcCell);
+    return tccMatrix;
+  }
+
+  private List<List<TitleBlockCell>> updateModelRowsDelete(int rowToDelete) {
+
+    tccMatrix.remove(rowToDelete);
+    rowsNumber -= 1;
+    return tccMatrix;
+  }
+
+  private List<List<TitleBlockCell>> updateModelColumnsDelete(int columnToDelete) {
+
+    for (int i = 0; i < rowsNumber; i++) {
+      tccMatrix.get(i).remove(columnToDelete);
+    }
+    return tccMatrix;
+  }
+
+  private List<List<TitleBlockCell>> createModel(int nrCol, int nrRows) {
+    String[] cellsNameAndContent = tableContent.split("\\+");

```

Comment:
also create a constant SEPARATOR and put + in it.

Reply:
Done